### ISSUE TITLE
Move if completed and delete data after delete torrent not working

### ISSUE THREAD

**Issue by Sveeeeeen** [rakshasa/rtorrent#1050]
Hello,
Is the wiki not up to date?
https://github.com/rakshasa/rtorrent/wiki/Common-Tasks-in-rTorrent#move-completed-torrents-to-a-fixed-location

I want to move my torrents when finished to a different dir.
```
method.insert = cfg.download,           private|const|string, (cat,(cfg.basedir),"download/")
method.insert = cfg.download_complete,  private|const|string, (cat,(cfg.basedir),"complete/")
```
Downloads should go do download/ and after finished to complete/.

```
method.insert = d.get_finished_dir, simple, "cat=$cfg.download_complete,$d.custom1="
method.insert = d.get_data_full_path, simple, "branch=((d.is_multi_file)),((cat,(d.directory))),((cat,(d.directory),/,(d.name)))"
method.insert = d.move_to_complete, simple, "d.directory.set=$argument.1=; execute=mkdir,-p,$argument.1=; execute=mv,-f,$argument.0=,$argument.1=; d.save_full_session="
method.set_key = event.download.finished,move_complete,"d.move_to_complete=$d.data_path=,$d.get_finished_dir="
```
This should work right? I just replaced my path in the first line.

But rtorrent is not moving my files. I am doing something wrong?

Even when delte the torrent using ruTorrent the files are stil existing and are not deleted.
`
method.set_key = event.download.erased,delete_erased,"execute=rm,-rf,--,$d.data_path="
`

I guess I am doing something terrible wrong. Please help me :D

**Comment by pyroscope** [comment]
`$cfg.download_complete=`, or use ().

**Comment by Sveeeeeen** [comment]
> `$cfg.download_complete=`, or use ().

Thanks.

But it still does not work. rtorrent still not moving my files.
Changed this line:

`method.insert = d.get_finished_dir, simple, "cat=$cfg.download_complete=,$d.custom1="`

**Comment by Sveeeeeen** [comment]
Ok for some reasons it is working now.

Just one problem. rTorrent is replacing slashes to "%2F".
Example:
download/LABEL/LABEL/MY_SHOW
after download completed it look like this
completed/LABEL%2FLABEL/MY_SHOW

Do I need to escape the slashes somehow?

**Comment by pyroscope** [comment]
no, rt does not, rut does. they are just stored that way (for no good reason).

**Comment by Sveeeeeen** [comment]
> no, rt does not, rut does. they are just stored that way (for no good reason).

But the destination is correct and rt does move the folder. Why it should be the problem of rut?

**Comment by topjes** [comment]
Hi, 

I had the same problem with "%2F" instead of subfolders. And I solved the problem.

I get out of the "autotools section" in rtorrent.rc completely
So I commented with "#" all these lines

<code># method.insert = d.get_finished_dir, simple, "cat=$cfg.download_complete,$d.custom1="
# method.insert = d.get_data_full_path, simple, "branch=((d.is_multi_file)),((cat,(d.directory))),((cat,(d.directory),/,(d.name)))"
# method.insert = d.move_to_complete, simple, "d.directory.set=$argument.1=; execute=mkdir,-p,$argument.1=; execute=mv,-f,$argument.0=,$argument.1=; d.save_full_session="
# method.set_key = event.download.finished,move_complete,"d.move_to_complete=$d.data_path=,$d.get_finished_dir="</code>

Then I used the rutorrent GUI Autotools plugin. And I configured there watch directory and all the rest.

And NO MORE #2F

**Comment by ahaghgoo** [comment]
Hi, I'm also getting the "%2F" when files are moved from 'temp' download folder to 'completed' download folder.  Is there a way to fix this without the use of AutoTools?  Thanks!

**Comment by kannibalox** [comment]
rTorrent has nothing to do with %2F getting inserted, check your ruTorrent configuration and get support through their channels.

**Comment by ahaghgoo** [comment]
Are you sure about that?  ruTorrent's AutoTools creates hard-linked files properly.  I'm having trouble with the way rTorrent is moving completed files.  Others have indicated that they experience the same thing.  Why are you so sure that it's not the case?  
--
Afshin Haghgoo
***@***.***

**Comment by kannibalox** [comment]
I am very sure about it, the root of the problem is that a URL-encoded value is getting stored in one of the custom variables, and then later move-on-completion scripts don't un-encode it. Custom values take whatever value is being sent, there's no way for rTorrent to silently URL encode it or anything.
```console
$ rtxmlrpc d.custom1 6A1B0461E42B8BEF716EF869844259CEFD1B0FD1

$ rtxmlrpc d.custom1.set 4A1B0461E42B8BEF716EF869847259CEFD1B0FD1 "/tmp/test/path"
$ rtxmlrpc d.custom1 4A1B0461E42B8BEF716EF869847259CEFD1B0FD1
/tmp/test/path
$ rtxmlrpc d.custom1.set 4A1B0461E42B8BEF716EF869847259CEFD1B0FD1 "/tmp%2Ftest%2Fpath"
$ rtxmlrpc d.custom1 4A1B0461E42B8BEF716EF869847259CEFD1B0FD1
/tmp%2Ftest%2Fpath
```
So far everyone experiencing this issue has indicated they are using ruTorrent, and there are even several tickets about this issue in their tracker:
- https://github.com/Novik/ruTorrent/issues/1439
- https://github.com/Novik/ruTorrent/issues/913

If you can provide [steps to reproduce](https://stackoverflow.com/help/minimal-reproducible-example) this issue with a new torrent on vanilla rTorrent (i.e. without ruTorrent installed), I'd be more than happy to test it out myself.

**Comment by ahaghgoo** [comment]
Ok this makes sense now.  I'm using AutoTools' 'watch' folder to import torrents, and then using rtorrents' file management to move files upon completion.  Thanks for clarifying this!  -A

**Comment by Jolly-Pirate** [comment]
Fix for the problem https://github.com/Jolly-Pirate/rtorrent-snippets/tree/master/fix-urlencoded-label
